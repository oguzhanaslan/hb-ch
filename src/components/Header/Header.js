import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';

const Header = () => {
  return (
    <header className='header col-xs-12'>
      <div className='col-xs-6'>
        <Link to='/'>Homepage</Link>
      </div>
      <div className='col-xs-6 text-right'>LinkVOTE Challenge</div>
    </header>
  );
};

export default Header;
