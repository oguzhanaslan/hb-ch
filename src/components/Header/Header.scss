.header {
  background: #fff;
  display: flex;
  align-content: center;
  padding: 30px 0;
}

.text-right {
  text-align: right;
}
