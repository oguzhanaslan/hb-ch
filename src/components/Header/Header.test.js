import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Header />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(2);
  });
});
