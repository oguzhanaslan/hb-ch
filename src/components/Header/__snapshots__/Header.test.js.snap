// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-6"
        >
          <Link
            to="/"
          >
            Homepage
          </Link>
        </div>,
        <div
          className="col-xs-6 text-right"
        >
          LinkVOTE Challenge
        </div>,
      ],
      "className": "header col-xs-12",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/"
          >
            Homepage
          </Link>,
          "className": "col-xs-6",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Homepage",
            "to": "/",
          },
          "ref": null,
          "rendered": "Homepage",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LinkVOTE Challenge",
          "className": "col-xs-6 text-right",
        },
        "ref": null,
        "rendered": "LinkVOTE Challenge",
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-6"
          >
            <Link
              to="/"
            >
              Homepage
            </Link>
          </div>,
          <div
            className="col-xs-6 text-right"
          >
            LinkVOTE Challenge
          </div>,
        ],
        "className": "header col-xs-12",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/"
            >
              Homepage
            </Link>,
            "className": "col-xs-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Homepage",
              "to": "/",
            },
            "ref": null,
            "rendered": "Homepage",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LinkVOTE Challenge",
            "className": "col-xs-6 text-right",
          },
          "ref": null,
          "rendered": "LinkVOTE Challenge",
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
